### PR TITLE
[Feature:Submission] seeking team message

### LINKED ISSUE #5654: Add an optional message string when "seeking a team/partner" for a gradeable
**What problem are you trying to solve with Submitty**
Currently students can press the "seek team/partner" button to add their contact information to the list of students who are looking for a partner.  This info is visible to other students in the course.

In some cases it would be helpful for the student to add additional information such as "I prefer Python" (if the assignment allows teams to pick their programming language) or "My local timezone is California" (if this is an online class with students in different timezones) or "I would like to do something related to ray tracing" (if students can pick their final project topic)

**Describe the way you'd like to solve this problem**

* let's add an optional message string input field next to seek team/partner
* let's store this string in the seek team database table
* let's display this string to the other students in the course

If we want to get fancy, we can add an instructor customizable "Seeking Team" message on the course settings page (similar to the "Office Hours Queue Welcome Message") that is displayed next to this input field.
The default string might be something like: "Optionally, provide your local timezone, desired project topic, or other information that would be relevant to forming your team."

### PR DESCRIPTION
**Pull request by jportorreal00**
### Please check if the PR fulfills these requirements:

* [ ] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?

Fixes #5654 : Messages were not an option.

### What is the new behavior?
Adds the option to have a seeking message when looking for a team or roommate

### Other information?

Tested by adding new messages or trying to enter empty messages.

### CODE REVIEW

File: migration/migrator/migrations/course/20200728162003_add_seek_message.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
```
Comment (mukul-kmr-jha) [inline migration/migrator/migrations/course/20200728162003_add_seek_message.py:4]: Please add these import statements
```suggestion
"""Migration for a given Submitty course database."""
import json
from pathlib import Path
def up(config, database, semester, course):
```

File: site/app/controllers/admin/ConfigurationController.php [hunk 2]
```diff
@@ -149,6 +151,8 @@ public function updateConfiguration(): MultiResponse {
                     'seating_only_for_instructor',
                     'queue_enabled',
                     'queue_contact_info',
+                    'seek_message_enabled',
+                    'seek_message_instructions',
```
Comment (mukul-kmr-jha) [inline site/app/controllers/admin/ConfigurationController.php:155]:
```suggestion
```

File: site/app/templates/submission/Team.twig [hunk 3]
```diff
@@ -105,28 +105,49 @@
...
             &nbsp;or&nbsp;<button class="btn btn-primary key_to_click" tabindex="0" onclick="location.href='{{ seek_url }}'">Seek Team/Partner </button>
+            <br>
         {% endif %}
+
+        {# Set Seeking Message #}
+        {% if config.isSeekMessageEnabled() %}
+            {% if seeking_partner %}
+                <br /> 
+                <br /> 
+                <p> {{entry['seek_message_instructions']}} </p>
```
Comment (mukul-kmr-jha) [inline site/app/templates/submission/Team.twig:119]:
```suggestion
                <p> {{ config.getSeekMessageInstructions() }}  </p>
```

File: site/app/libraries/database/DatabaseQueries.php [hunk 4]
```diff
@@ -2782,6 +2783,36 @@ public function removeFromSeekingTeam($g_id, $user_id) {
...
+
+    /**
+     * Edit the user's message from table seeking_team
+     *
+     * @param string $g_id
+     * @param string $user_id
+     * @return string $user_message
+     */
+    public function getSeekMessageByUserId($g_id, $user_id) {
+        $this->course_db->query(
```
Comment (mukul-kmr-jha) [inline site/app/libraries/database/DatabaseQueries.php:2805]: Update the php-docs.

Comment (jportorreal00) [reply site/app/libraries/database/DatabaseQueries.php]: I really liked the suggestions, I think I have them implemented right. The user can now set a message. If one is set then you can edit the message in the text area or remove it entirely.

File: site/app/templates/submission/Team.twig [hunk 5]
```diff
@@ -105,28 +105,62 @@
...
+        {# Set Seeking Message #}
+        {% if seeking_enabled %}
+            {% if seeking_partner %}
+                <br /> 
+                <br /> 
+                <p> {{ seeking_instructions }}  </p>
+                <br />
+                {% if user.getSeekMessage(gradeable.getId()) != "N/A" %}
+                    <form action="{{ set_message_url }}" method="post">
+                        <textarea id="seek_message" class="autoExpand" style="overflow:hidden;" rows='2' cols='30' placeholder="Seeking Message">{{ user.getSeekMessage(gradeable.getId()) }}</textarea>
```
Comment (mukul-kmr-jha) [inline site/app/templates/submission/Team.twig:123]:
```suggestion
                        <textarea name="seek_message" class="autoExpand" style="overflow:hidden;" rows='2' cols='30' placeholder="Seeking Message">{{ user.getSeekMessage(gradeable.getId()) }}</textarea>
```

**Comment by jportorreal00** [comment]
@mukul-kmr-jha thank you so much for the suggestions! they're always super helpful :)

**Pull request merged**